### PR TITLE
SOF-7437: default workflow for spin magn and soc bandstructure calc

### PR DESCRIPTION
**Pull request by pranabdas**

### CODE REVIEW

File: assets/subworkflows/espresso/soc_bands.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Spin orbit coupling bandstructure
```
Comment (timurbazhirov) [inline assets/subworkflows/espresso/soc_bands.yml:1]: band_structure_soc

File: assets/subworkflows/espresso/spin_magn_bands.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Spin magnetic bandstructure
```
Comment (timurbazhirov) [inline assets/subworkflows/espresso/spin_magn_bands.yml:1]: band_structure_magn

File: assets/workflows/espresso/soc_bands.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+name: Spin orbit coupling bandstructure - QE
```
Comment (timurbazhirov) [inline assets/workflows/espresso/soc_bands.yml:1]: Same as above

File: assets/subworkflows/espresso/band_structure_soc.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    functions:
+      head: true
+    type: executionBuilder
+  - config:
+      execName: pw.x
+      flavorName: pw_bands_soc
+      name: pw_bands_soc
+    type: executionBuilder
+  - config:
+      execName: bands.x
```
Comment (pranabdas) [inline assets/subworkflows/espresso/band_structure_soc.yml:23]: After https://github.com/Exabyte-io/application-flavors/pull/127 is merged, add two separate bands.x units for up and down spin components.

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**